plugins/analytics/public/threatminer.py

```diff
@@ -207,83 +177,47 @@ def analyze(observable, results):
         links = set()
         params = {"q": observable.value, "rt": 3}
         json_result = ThreatMinerApi.fetch(observable, params, "sample.php")
-        if json_result:
-            json_string = json.dumps(
-                json_result,
-                sort_keys=True,
-                indent=4,
-                separators=(",", ": ")
-            )
-
-            result = dict(
-                [
-                    ("raw", json_string),
-                    ("source", "threatminer_query")
-                ]
-            )
-            results.update(raw=json_string)
-            _results = json_result.get("results")
+        _results, result = aux_checker(json_result)
+        if not _results:
+            return []
+
+        for r in _results:
+            _hosts = r.get("hosts")
+            if _hosts:
+                for ip in _hosts:
+                    try:
+                        o_ip = Ip.get_or_create(value=ip)
+                        o_ip.tag(observable.get_tags())
+                        links.update(o_ip.active_link_to(observable,
+                            "seen connecting to", "ThreatMiner")
+                        )
+                    except ObservableValidationError as e:
+                        logging.error("Caught an exception: {}".format(e))
 
-            if not _results:
-                return list(links)
+            _domains = r.get("domains")
+            if not _domains:
```

Comment:
You're right, this guard statement is not necessary here :)

You can also do like for `_hosts`, 

```python
for domain in r.get("domains", []):
  try:
    ...
```